apps/src/lib/node/ledger/protocol/mod.rs

```diff
@@ -236,30 +206,39 @@ where
     H: 'static + StorageHasher + Sync,
     CA: 'static + WasmCacheAccess + Sync,
 {
-    let verifiers_addr = verifiers
-        .iter()
-        .map(|(addr, _, _)| addr)
-        .cloned()
-        .collect::<BTreeSet<_>>();
-
     verifiers
         .par_iter()
-        .try_fold(VpsResult::default, |mut result, (addr, keys, vp)| {
+        // TODO temporary pending on <https://github.com/anoma/anoma/issues/193>
+        .filter(|addr| !matches!(addr, Address::Implicit(_)))
+        .try_fold(VpsResult::default, |mut result, addr| {
             let mut gas_meter = VpGasMeter::new(initial_gas);
-            let accept = match &vp {
-                Vp::Wasm(vp) => wasm::run::vp(
-                    vp,
-                    tx,
-                    addr,
-                    storage,
-                    write_log,
-                    &mut gas_meter,
-                    keys,
-                    &verifiers_addr,
-                    vp_wasm_cache.clone(),
-                )
-                .map_err(Error::VpRunnerError),
-                Vp::Native(internal_addr) => {
+            let accept = match &addr {
+                Address::Established(_) => {
+                    let (vp, gas) = storage
+                        .validity_predicate(addr)
```

Comment:
note here that the VP loading (and compiling, if it's not cached already) was done sequentially in a single-thread before (in `check_vps`), but now we'll be loading them inside the parallel iterator